src/qibocal/calibrations/characterization/resonator_spectroscopy.py

```diff
@@ -87,12 +91,31 @@ def resonator_spectroscopy(
     )
 
     # repeat the experiment as many times as defined by software_averages
-    count = 0
     for iteration in range(software_averages):
-        # sweep the parameter
-        for delta_freq in delta_frequency_range:
+        results = platform.sweep(sequence, sweeper, nshots=nshots)
+
+        while (
+            any(result.in_progress for result in results.values())
+            or len(fast_sweep_data) == 0
+        ):
+            # retrieve the results for every qubit
+            for qubit in qubits:
+                # average msr, phase, i and q over the number of shots defined in the runcard
+                result = results[ro_pulses[qubit].serial]
+                # store the results
+                r = result.to_dict()
```

Comment:
The code will fail again if you don't specify the `average` is `False`.
By default `to_dict` performs the average for i and q.
```suggestion
                r = result.to_dict(average=False)
```